### PR TITLE
ARROW-12 Support ObjectId datatype

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: bindings/python/test/test_bson.py [hunk 1]
```diff
@@ -18,12 +18,12 @@
...
 from pymongoarrow.lib import process_bson_stream
 from pymongoarrow.schema import Schema
-from pymongoarrow.types import int32, int64
+from pymongoarrow.types import int64, ObjectId
 
 
 class TestBsonToArrowConversionBase(TestCase):
     def setUp(self):
-        self.schema = Schema({'_id': int32(),
+        self.schema = Schema({'_id': ObjectId,
```
Comment (ShaneHarvey) [inline bindings/python/test/test_bson.py:26]: Are we now loosing test coverage for int32?

File: bindings/python/pymongoarrow/types.py [hunk 2]
```diff
@@ -64,8 +71,13 @@ def _get_internal_typemap(typemap):
     internal_typemap = {}
     for fname, ftype in typemap.items():
         for checker, internal_id in _TYPE_CHECKER_TO_INTERNAL_TYPE.items():
-            if checker(ftype):
-                internal_typemap[fname] = internal_id
-                continue
+            # Catch error where the pyarrow checkers are looking for an `id`
+            # attribute that might not exist on non-pyarrow types
+            # (like ObjectId).
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/types.py:76]: I don't follow this comment. Can you give an example?

File: bindings/python/test/test_bson.py (hunk 1 above)
Comment (blink1073) [reply bindings/python/test/test_bson.py:26]: All of the arrow types are checked in `test_pandas.py`

File: bindings/python/pymongoarrow/types.py (hunk 2 above)
Comment (blink1073) [reply bindings/python/pymongoarrow/types.py]: Updated

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Existing changes LGTM! One last thing is to update the docs for the supported data types: https://github.com/mongodb-labs/mongo-arrow/blob/main/bindings/python/docs/source/supported_types.rst and the changelog.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**